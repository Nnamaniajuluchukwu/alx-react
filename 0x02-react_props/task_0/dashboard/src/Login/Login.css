
.App-body {
    font-size: 1rem;
    font-family: var(--font-family-general);
    margin: auto;
    height: 23rem;
    padding: 3rem;
}
  
.App-body input {
    margin-right: 5px;
}
