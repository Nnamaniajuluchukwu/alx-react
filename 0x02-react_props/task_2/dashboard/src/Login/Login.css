
.App-body {
    border-bottom: 3px solid #e0354b;
    font-size: 1rem;
    font-family: var(--font-family-general);
    margin: auto;
    height: 23rem;
    padding: 3rem;
}
  
.App-body input {
    margin-right: 5px;
}
