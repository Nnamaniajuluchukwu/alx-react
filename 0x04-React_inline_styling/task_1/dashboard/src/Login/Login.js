import React from "react";
import { StyleSheet, css } from "aphrodite";


const styles = StyleSheet.create({
    container: {
        fontSize: '1rem',
        fontFamily: 'Arial, Helvetica, sans-serif',
        margin: 'auto',
        height: '13rem',
        padding: '3rem'
    },
    input: {
        marginRight: '5px'
    }
});

export default function Login() {
    return (
        <>
            <div className={css(styles.container)}>
                <p>Login to access the full dashboard</p>
                <label htmlFor='email'>Email </label>
                <input className={css(styles.input)} type='email' id='email' autoComplete="on"></input>
                <label htmlFor='password'> Password </label>
                <input type='password' id='password' autoComplete="on"></input>
                <button type='submit'>OK</button>
            </div>
        </>
    );
}
