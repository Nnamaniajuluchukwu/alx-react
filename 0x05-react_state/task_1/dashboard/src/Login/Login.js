import React, {Component} from "react";
import { StyleSheet, css } from "aphrodite";


const styles = StyleSheet.create({
    container: {
        fontSize: '1rem',
        fontFamily: 'Arial, Helvetica, sans-serif',
        margin: 'auto',
        height: '13rem',
        padding: '3rem',
        "@media (max-width: 900px)": {
            display: "flex",
            flexDirection: "column",
        },
    },
    input: {
        marginRight: '5px'
    }
});

export default class Login extends Component {
    constructor(props) {
        super(props);
        this.state = {
            isLoggedIn: false,
            email: '',
            password: '',
            enableSubmit: false
        }
        this.handleLoginSubmit = this.handleLoginSubmit.bind(this);
        this.handleChangeEmail = this.handleChangeEmail.bind(this);
        this.handleChangePassword = this.handleChangePassword.bind(this);
    }

    handleLoginSubmit() {
        this.setState({
            isLoggedIn: true
        });
    }

    handleChangeEmail(e) {
        const email = e.target.value;
        this.setState({ email }, () => this.updateEnableSubmit());
      }

    handleChangePassword(e) {
        const password = e.target.value;
        this.setState({ password }, () => this.updateEnableSubmit());
    }

    updateEnableSubmit() {
        const { email, password } = this.state;
        const enableSubmit = email !== '' && password !== '';
        this.setState({ enableSubmit });
    }

    render() {
        return (
            <>
                <div className={css(styles.container)}>
                    <form>
                        <p>Login to access the full dashboard</p>
                        <label htmlFor='email'>Email: </label>
                        <input
                            className={css(styles.input)}
                            type='email'id='email' autoComplete="on"
                            value={this.state.email} onChange={this.handleChangeEmail}>
                        </input>
                        <label htmlFor='password'> Password: </label>
                        <input
                            type='password' id='password' autoComplete="on"
                            value={this.state.password} onChange={this.handleChangePassword}></input>
                        <input type='submit' onSubmit={this.handleLoginSubmit} disabled={!this.state.enableSubmit}></input>
                    </form>
                </div>
            </>
        );
    }
}

// Function Method 
// export default function Login() {
//     const [email, setEmail] = useState('');
//     const [password, setPassword] = useState('');
//     const [isLoggedIn, setIsLoggedIn] = useState(false);
//     const [enableSubmit, setEnableSubmit] = useState(false);

//     const handleLoginSubmit = () => {
//         setIsLoggedIn(true);
//     };

//     const handleChangeEmail = (e) => {
//         setEmail(e.target.value);
//         updateEnableSubmit(e.target.value, password);
//     };

//     const handleChangePassword = (e) => {
//         setPassword(e.target.value);
//         updateEnableSubmit(email, e.target.value);
//     };

//     const updateEnableSubmit = (newEmail, newPassword) => {
//         setEnableSubmit(newEmail !== "" && newPassword !== "");
//     };

//     return (
//         <>
//             <div className={css(styles.container)}>
//                 <form>
//                     <p>Login to access the full dashboard</p>
//                     <label htmlFor='email'>Email: </label>
//                     <input
//                         className={css(styles.input)}
//                         type='email'id='email' autoComplete="on"
//                         value={email} onChange={handleChangeEmail}>
//                     </input>
//                     <label htmlFor='password'> Password: </label>
//                     <input
//                         type='password' id='password' autoComplete="on"
//                         value={password} onChange={handleChangePassword}></input>
//                     <input type='submit' onSubmit={handleLoginSubmit} disabled={!enableSubmit}></input>
//                 </form>
//             </div>
//         </>
//     );
// }
