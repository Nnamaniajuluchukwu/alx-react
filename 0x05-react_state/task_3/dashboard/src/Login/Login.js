import React, {Component} from "react";
import { StyleSheet, css } from "aphrodite";


const styles = StyleSheet.create({
    container: {
        fontSize: '1rem',
        fontFamily: 'Arial, Helvetica, sans-serif',
        margin: 'auto',
        height: '13rem',
        padding: '3rem',
        "@media (max-width: 900px)": {
            display: "flex",
            flexDirection: "column",
        },
    },
    input: {
        marginRight: '5px'
    }
});

export default class Login extends Component {
    constructor(props) {
        super(props);
        this.state = {
            email: '',
            password: '',
            enableSubmit: false
        }
        this.handleLoginSubmit = this.handleLoginSubmit.bind(this);
        this.handleChangeEmail = this.handleChangeEmail.bind(this);
        this.handleChangePassword = this.handleChangePassword.bind(this);
    }

    handleLoginSubmit() {
        this.props.logIn(this.state.email, this.state.password);
    }

    handleChangeEmail(e) {
        const email = e.target.value;
        this.setState({ email }, () => this.updateEnableSubmit());
      }

    handleChangePassword(e) {
        const password = e.target.value;
        this.setState({ password }, () => this.updateEnableSubmit());
    }

    updateEnableSubmit() {
        const { email, password } = this.state;
        const enableSubmit = email !== '' && password !== '';
        this.setState({ enableSubmit });
    }

    render() {
        return (
            <>
                <div className={css(styles.container)}>
                    <form>
                        <p>Login to access the full dashboard</p>
                        <label htmlFor='email'>Email: </label>
                        <input
                            className={css(styles.input)}
                            type='email'id='email' autoComplete="on"
                            value={this.state.email} onChange={this.handleChangeEmail}>
                        </input>
                        <label htmlFor='password'> Password: </label>
                        <input
                            type='password' id='password' autoComplete="on"
                            value={this.state.password} onChange={this.handleChangePassword}></input>
                        <input type='submit' onClick={this.handleLoginSubmit} disabled={!this.state.enableSubmit}></input>
                    </form>
                </div>
            </>
        );
    }
}

// Function Method 
// export default function Login({ logIn }) {
//     const [email, setEmail] = useState('');
//     const [password, setPassword] = useState('');
//     const [enableSubmit, setEnableSubmit] = useState(false);

//     const handleLoginSubmit = () => {
//         logIn(email, password);
//     };

//     const handleChangeEmail = (e) => {
//         setEmail(e.target.value);
//         updateEnableSubmit(e.target.value, password);
//     };

//     const handleChangePassword = (e) => {
//         setPassword(e.target.value);
//         updateEnableSubmit(email, e.target.value);
//     };

//     const updateEnableSubmit = (newEmail, newPassword) => {
//         setEnableSubmit(newEmail !== "" && newPassword !== "");
//     };

//     return (
//         <>
//             <div className={css(styles.container)}>
//                 <form>
//                     <p>Login to access the full dashboard</p>
//                     <label htmlFor='email'>Email: </label>
//                     <input
//                         className={css(styles.input)}
//                         type='email'id='email' autoComplete="on"
//                         value={email} onChange={handleChangeEmail}>
//                     </input>
//                     <label htmlFor='password'> Password: </label>
//                     <input
//                         type='password' id='password' autoComplete="on"
//                         value={password} onChange={handleChangePassword}></input>
//                     <input type='submit' onClick={handleLoginSubmit} disabled={!enableSubmit}></input>
//                 </form>
//             </div>
//         </>
//     );
// }
