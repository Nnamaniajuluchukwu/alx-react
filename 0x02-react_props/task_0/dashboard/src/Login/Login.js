import React from "react";
import './Login.css';


export default function Login() {
    return (
        <>
            <body className="App-body">
                <p>Login to access the full dashboard</p>
                <label for='email'>Email </label>
                <input type='email' id='email'></input>
                <label for='password'> Password </label>
                <input type='password' id='password'></input>
                <button type='submit'>OK</button>
            </body>
        </>
    );
}
