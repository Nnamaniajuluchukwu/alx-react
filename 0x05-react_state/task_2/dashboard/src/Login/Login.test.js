import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Login from './Login';
import { StyleSheetTestUtils } from 'aphrodite';


describe('Login Component', () => {
  let wrapper;
  beforeAll(() => {
      wrapper = shallow(<Login />);
      StyleSheetTestUtils.suppressStyleInjection();
  })

  afterAll(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('should render without crashing', () => {
      expect(wrapper.exists()).toBeTruthy();
  });

  it('should contains 3 inputs', () => {
      const input = wrapper.find('input');
      expect(input.length).toEqual(3);
  });

  it('should contains 2 labels', () => {
      const label = wrapper.find('label');
      expect(label.length).toEqual(2);
  });
  
  it('should verify that the submit button is disabled by default', () => {
      const wrapper = shallow(<Login />);
      expect(wrapper.find('input').at(2).prop('disabled')).toBeTruthy();
  });
  
  it('should verify that after changing the value of the two inputs, the button is enabled', () => {
    wrapper.find('input').at(0).simulate('change', { target: { email: 'Eyang' } });
    wrapper.find('input').at(1).simulate('change', { target: { password: 'alx-react' } });
    expect(wrapper.find('input').at(2).prop('disabled')).toBeFalsy();
  });

  it('should match the snapshot', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
  });
});
