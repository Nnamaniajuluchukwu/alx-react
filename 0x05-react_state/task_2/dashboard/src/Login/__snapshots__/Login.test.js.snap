// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component should match the snapshot 1`] = `
<Fragment>
  <div
    className="container_ae45qw"
  >
    <form>
      <p>
        Login to access the full dashboard
      </p>
      <label
        htmlFor="email"
      >
        Email: 
      </label>
      <input
        autoComplete="on"
        className="input_m0te3e"
        id="email"
        onChange={[Function]}
        type="email"
      />
      <label
        htmlFor="password"
      >
         Password: 
      </label>
      <input
        autoComplete="on"
        id="password"
        onChange={[Function]}
        type="password"
      />
      <input
        disabled={false}
        onClick={[Function]}
        type="submit"
      />
    </form>
  </div>
</Fragment>
`;
